### PR TITLE
Show admin console header if serverinfo is forbidden

### LINKED ISSUE #30683: Infinite loader on the admin console for non-admin users
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

When a non-admin user tries to access the admin console, after logging in, they end up on a page with an infinite loader.

### Version

nightly

### Regression

- [ ] The issue is a regression

### Expected behavior

The admin console should display an error message after login.

### Actual behavior

The admin console shows an infinite loader after login.

### How to Reproduce?

Create a user without roles on the real `master` and log in to the admin console.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by hmlnarik**
@ssilvert Could you please check this is the solution you were anticipating?

Fixes: #30683

### CODE REVIEW

File: js/apps/admin-ui/src/context/server-info/ServerInfoProvider.tsx [hunk 1]
```diff
@@ -22,7 +23,15 @@ export const ServerInfoProvider = ({ children }: PropsWithChildren) => {
   const { adminClient } = useAdminClient();
   const [serverInfo, setServerInfo] = useState<ServerInfoRepresentation>();
 
-  useFetch(adminClient.serverInfo.find, setServerInfo, []);
+  useFetch(adminClient.serverInfo.find, setServerInfo, [], (error) => {
```
[the remark this answers is not in the document]
Comment (hmlnarik) [reply js/apps/admin-ui/src/context/server-info/ServerInfoProvider.tsx:26]: Good idea, thanks.

File: .github/workflows/js-ci.yml [hunk 2]
```diff
@@ -282,7 +282,7 @@ jobs:
           wait-on: http://localhost:8080
           working-directory: js/apps/admin-ui
         env:
-          CYPRESS_BASE_URL: http://localhost:8080/admin/
+          #CYPRESS_BASE_URL: http://localhost:8080/admin/
```
Comment (jonkoops) [inline .github/workflows/js-ci.yml:285]: Probably committed by accident?
```suggestion
          CYPRESS_BASE_URL: http://localhost:8080/admin/
```

File: js/apps/admin-ui/cypress/e2e/masthead_test.spec.ts [hunk 3]
```diff
@@ -64,6 +66,48 @@ describe("Masthead tests", () => {
...
   });
 
+  describe("Login works for unprivileged users", () => {
+    const realmName = `test-realm-${uuid()}`;
+    const username = `test-user-${uuid()}`;
+
+    before(async () => {
+      await adminClient.createRealm(realmName, { enabled: true });
+
+      console.info("Creating user" + username);
```
Comment (jonkoops) [inline js/apps/admin-ui/cypress/e2e/masthead_test.spec.ts:76]: Stray console log?
```suggestion
```

File: js/apps/admin-ui/cypress/e2e/masthead_test.spec.ts [hunk 4]
```diff
@@ -64,6 +66,48 @@ describe("Masthead tests", () => {
...
+      await adminClient.inRealm(realmName, () =>
+        adminClient.createUser({
```
Comment (jonkoops) [inline js/apps/admin-ui/cypress/e2e/masthead_test.spec.ts:78]: Should the promise from `createUser` not be returned? This way `inRealm` can await it.
```suggestion
        return adminClient.createUser({
```

File: .github/workflows/js-ci.yml (hunk 2 above)
Comment (hmlnarik) [reply .github/workflows/js-ci.yml:285]: No, this is intentional. Without it, the https://github.com/keycloak/keycloak/actions/runs/9710664453?pr=30893 shows that the `/admin` is included twice.

File: js/apps/admin-ui/cypress/e2e/masthead_test.spec.ts (hunk 3 above)
Comment (hmlnarik) [reply js/apps/admin-ui/cypress/e2e/masthead_test.spec.ts:76]: This is draft change.

File: js/apps/admin-ui/cypress/e2e/masthead_test.spec.ts (hunk 4 above)
Comment (hmlnarik) [reply js/apps/admin-ui/cypress/e2e/masthead_test.spec.ts:78]: Since this is not a block but a lambda, it is done implicitly.

Comment (jonkoops) [reply js/apps/admin-ui/cypress/e2e/masthead_test.spec.ts:77]: It is actually not if you put brackets around the body, see the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#function_body).

**Review by ssilvert — APPROVED** [review]
I tried it out.  Looks like a solid solution.

File: js/apps/admin-ui/cypress/e2e/masthead_test.spec.ts (hunk 4 above)
Comment (hmlnarik) [reply js/apps/admin-ui/cypress/e2e/masthead_test.spec.ts:77]: There are no brackets. Thus it is expression, not a statement.

Comment (jonkoops) [reply js/apps/admin-ui/cypress/e2e/masthead_test.spec.ts:77]: Ah sorry, I misread.

**Pull request merged**